hive-metastore/src/main/java/org/apache/iceberg/hive/HiveSchemaConverter.java

```diff
@@ -64,7 +64,7 @@ static Type convert(TypeInfo typeInfo, boolean autoConvert) {
     List<Types.NestedField> result = new ArrayList<>(names.size());
     for (int i = 0; i < names.size(); ++i) {
       result.add(Types.NestedField.optional(id++, names.get(i), convertType(typeInfos.get(i)),
-          comments.isEmpty() ? null : comments.get(i)));
+          comments.isEmpty() || i >= comments.size() ? null : comments.get(i)));
```

Comment:
Nit: Possibly add surrounding braces since this ternary is inline as an argument.

Reply:
fix thanks @kbendick